### ISSUE TITLE
All commands erroring out with nil:NilClass

### ISSUE THREAD

**Issue by mcmillion** [castwide/solargraph#86]
Hi! I'm trying to get this configured in neovim. I've got the latest gem installed (0.26.1) and I'm using this configuration in neovim with LanguageClient-neovim:

```
let g:LanguageClient_serverCommands = {
\ 'javascript': ['javascript-typescript-stdio'],
\ 'javascript.jsx': ['javascript-typescript-stdio'],
\ 'ruby': ['solargraph', 'stdio'],
\ }
```

I know LanguageClient-neovim is working, because the JavaScript side of things works. However, on the Ruby side, no commands seem to work and they all complain about nils. Some examples of errors I'm seeing:

Trying go hover or goto defintion: `[NoMethodError] undefined method 'definitions_at' for nil:NilClass`
Trying to rename: `[NoMethodError] undefined method 'references_from' for nil:NilClass`
Trying to format: `[NoMethodError] undefined method 'read_text' for nil:NilClass`

**Comment by castwide** [comment]
Can you confirm that the client is sending the `initialize` method before it starts sending other requests? Those errors can happen if the server hasn't created a library yet.

This problem might also be related to a threading bug that's been resolved in the master branch. If that's the case, it should be fixed in gem 0.27.0, which I expect to release on Monday.

**Comment by mcmillion** [comment]
> Can you confirm that the client is sending the initialize method before it starts sending other requests?

I'm pretty new to language servers. How would I go about checking that?

**Comment by castwide** [comment]
> I'm pretty new to language servers. How would I go about checking that?

I'm not familiar enough with Neovim to say. I've installed it on a CentOS box so I can test with it, but haven't had time to dig into it yet.

Gem version 0.27.0 is published. Among other things, the language server will use a default library if one hasn't been constructed in response to an `initialize` method. This should fix the nil error with uninitialized servers, although I expect to add a warning when it happens since it violates the LSP specification.

Improving the language server's logging/debugging capability is on the roadmap. This should help with troubleshooting clients in the future.

**Comment by castwide** [comment]
Sorry, I jumped the gun. v0.27.0 didn't include the default library. I went ahead and published it in v0.27.1.

**Comment by mcmillion** [comment]
Hey! 0.27.1 fixed the issues I was having. Fantastic!
